File: src/main/java/yummyloop/example/item/Item.java
package yummyloop.example.item;

import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;
import net.minecraft.item.ItemGroup;
//import net.minecraft.item.Item;

public class Item extends net.minecraft.item.Item {
           
    public Item(net.minecraft.item.Item.Settings itemSettings) {
        super(itemSettings);
    }

    public Item() {
        super(new net.minecraft.item.Item.Settings().group(ItemGroup.MISC));
    }

    public Item(int maxDamage, int maxCount, net.minecraft.item.Item recipeRemainder, ItemGroup group, net.minecraft.util.Rarity rarity, net.minecraft.item.FoodComponent foodComponent) {
        super(new net.minecraft.item.Item.Settings().maxDamage(maxDamage).maxCount(maxCount).recipeRemainder(recipeRemainder).group(group).rarity(rarity).food(foodComponent));
    }
 
    public Item(int maxDamage, int maxCount, ItemGroup group) {
        super(new net.minecraft.item.Item.Settings().maxDamage(maxDamage).maxCount(maxCount).group(group));
    }

    public Item(ItemGroup group) {
        super(new net.minecraft.item.Item.Settings().group(group));
    }

    public Item(String modid, String itemName){
        super(new net.minecraft.item.Item.Settings().group(ItemGroup.MISC));
        register(modid, itemName);
    }

    public Item(String modid, String itemName, ItemGroup group) {
        super(new net.minecraft.item.Item.Settings().group(group));
        register(modid, itemName);
    }

    public boolean register (String modid, String itemName) {
        Registry.register(Registry.ITEM, new Identifier(modid, itemName), this);
        return true;
    }
    

}

File: src/main/yummyloop/example/item/Item.kt
package yummyloop.example.item

import net.minecraft.client.item.TooltipContext
import net.minecraft.item.ItemStack
import net.minecraft.text.Text
import net.minecraft.text.TranslatableText
import net.minecraft.world.World
import yummyloop.example.registry.RegistryManager
import net.minecraft.item.Item as VanillaItem
import net.minecraft.item.ItemGroup as VanillaItemGroup
g
open class Item(itemName: String, settings : Settings) : VanillaItem(settings) {
    private val tooltip = ArrayList<Text>()

    init {
        register(itemName)
    }
    constructor(itemName : String) :
            this(itemName, Settings().group(VanillaItemGroup.MISC))
    constructor(itemName : String, group : VanillaItemGroup?) :
            this(itemName, Settings().group(group))
    constructor(itemName : String, group : VanillaItemGroup?, maxCount : Int) :
            this(itemName, Settings().group(group).maxCount(maxCount))

    private fun register (itemName : String) {
        RegistryManager.register(this, itemName)
    }

    override fun appendTooltip(itemStack: ItemStack?, world: World?, tooltip: MutableList<Text>?, tooltipContext: TooltipContext?) {
        tooltip?.addAll(this.tooltip)
    }

    /**
     * Appends a tooltip to the item
     * @param tooltip example: item.tutorial.fabric_item.tooltip
     */
    fun addTooltip(tooltip : String) : Item{
        this.tooltip.add(TranslatableText(tooltip))
        return this
    }
}